eZ/Publish/Core/Repository/SiteAccessAware/RoleService.php

```diff
@@ -0,0 +1,215 @@
+<?php
+
+/**
+ * RoleService class.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Publish\Core\Repository\SiteAccessAware;
+
+use eZ\Publish\API\Repository\RoleService as RoleServiceInterface;
+use eZ\Publish\API\Repository\Values\User\Limitation\RoleLimitation;
+use eZ\Publish\API\Repository\Values\User\Policy;
+use eZ\Publish\API\Repository\Values\User\PolicyCreateStruct;
+use eZ\Publish\API\Repository\Values\User\PolicyDraft;
+use eZ\Publish\API\Repository\Values\User\PolicyUpdateStruct;
+use eZ\Publish\API\Repository\Values\User\Role;
+use eZ\Publish\API\Repository\Values\User\RoleAssignment;
+use eZ\Publish\API\Repository\Values\User\RoleCreateStruct;
+use eZ\Publish\API\Repository\Values\User\RoleDraft;
+use eZ\Publish\API\Repository\Values\User\RoleUpdateStruct;
+use eZ\Publish\API\Repository\Values\User\User;
+use eZ\Publish\API\Repository\Values\User\UserGroup;
+
+/**
+ * RoleService for SiteAccessAware layer.
+ *
+ * Currently does nothing but hand over calls to aggregated service.
```

Comment:
Out of scope: maybe we should introduce the base class for services decorators? The main benefit is avoiding code duplication in each PAPI layer. Even right now implementation of struct factory methods  (e.g. `\eZ\Publish\API\Repository\RoleService::newRoleCreateStruct`) are identical on both SignalSlot and SiteaccessAware layers. WDYT?

The similar practice is used in Doctrine:   https://github.com/doctrine/doctrine2/blob/master/lib/Doctrine/ORM/Decorator/EntityManagerDecorator.php

Reply:
hmm, maybe. At least if we end up doing more layers in the feature _(permissions + cache for instance, they will do the same thing on these once)_.

Note: If we don't intend to fully translate Roles, we can also choose to ommit this class completely like I opted with Field-, Url- and UrlWildcard-Service.